components/director/internal/tenantmapping/mapper_for_user.go

```diff
@@ -6,55 +6,110 @@ import (
 	"strings"
 
 	"github.com/kyma-incubator/compass/components/director/internal/consumer"
-
 	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
 	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
 )
 
-func NewMapperForUser(staticUserRepo StaticUserRepository, tenantRepo TenantRepository) *mapperForUser {
+func NewMapperForUser(staticUserRepo StaticUserRepository, staticGroupRepo StaticGroupRepository, tenantRepo TenantRepository) *mapperForUser {
 	return &mapperForUser{
-		staticUserRepo: staticUserRepo,
-		tenantRepo:     tenantRepo,
+		staticUserRepo:  staticUserRepo,
+		staticGroupRepo: staticGroupRepo,
+		tenantRepo:      tenantRepo,
 	}
 }
 
 type mapperForUser struct {
-	staticUserRepo StaticUserRepository
-	tenantRepo     TenantRepository
+	staticUserRepo  StaticUserRepository
+	staticGroupRepo StaticGroupRepository
+	tenantRepo      TenantRepository
+}
+
+func stringInSlice(a string, list []string) bool {
+	for _, b := range list {
+		if b == a {
+			return true
+		}
+	}
+	return false
+}
+
+const tenantGroupPrefix = "tenantID="
+
+// GetGroupScopesAndTenant get all scopes from group array, without duplicates
+func GetGroupScopesAndTenant(groups []StaticGroup) (string, string) {
+	var filteredScopesArray []string
+	var tenantID string
+
+	for _, group := range groups {
+		if strings.HasPrefix(group.GroupName, tenantGroupPrefix) {
+			// group name is tenantID
```

Comment:
you don't need such comments, you can extract this 'if logic' to some func with a descriptive name